src/main/java/bio/terra/service/duos/DuosService.java

```diff
@@ -1,21 +1,76 @@
 package bio.terra.service.duos;
 
+import bio.terra.common.ExceptionUtils;
 import bio.terra.model.DuosFirecloudGroupModel;
+import bio.terra.model.RepositoryStatusModelSystems;
+import bio.terra.service.auth.iam.IamRole;
 import bio.terra.service.auth.iam.IamService;
 import bio.terra.service.auth.iam.exception.IamConflictException;
+import bio.terra.service.duos.exception.DuosDatasetBadRequestException;
+import bio.terra.service.duos.exception.DuosDatasetNotFoundException;
+import bio.terra.service.duos.exception.DuosFirecloudGroupUpdateConflictException;
 import com.google.common.annotations.VisibleForTesting;
+import java.time.Instant;
+import java.util.List;
 import java.util.UUID;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.dao.CannotSerializeTransactionException;
 import org.springframework.stereotype.Component;
 
 @Component
 public class DuosService {
   private static final Logger logger = LoggerFactory.getLogger(DuosService.class);
+  @VisibleForTesting static final boolean IS_CRITICAL_SYSTEM = false;
   private final IamService iamService;
+  private final DuosClient duosClient;
+  private final DuosDao duosDao;
 
-  public DuosService(IamService iamService) {
+  public DuosService(IamService iamService, DuosClient duosClient, DuosDao duosDao) {
     this.iamService = iamService;
+    this.duosClient = duosClient;
+    this.duosDao = duosDao;
+  }
+
+  /**
+   * @return status of DUOS and its subsystems
+   */
+  public RepositoryStatusModelSystems status() {
+    try {
+      SystemStatus status = duosClient.status();
+      return new RepositoryStatusModelSystems()
+          .ok(status.ok())
+          .critical(IS_CRITICAL_SYSTEM)
+          .message(status.systems().toString());
+    } catch (Exception ex) {
+      String errorMsg = "DUOS status check failed";
+      logger.error(errorMsg, ex);
+      return new RepositoryStatusModelSystems()
+          .ok(false)
+          .critical(IS_CRITICAL_SYSTEM)
+          .message(errorMsg + ": " + ExceptionUtils.formatException(ex));
+    }
+  }
+
+  /**
+   * @return all TDR-managed DUOS Firecloud groups
+   */
+  public List<DuosFirecloudGroupModel> retrieveFirecloudGroups() {
+    return duosDao.retrieveFirecloudGroups();
+  }
+
+  /**
+   * @param duosId DUOS dataset ID
+   * @return TDR-managed DUOS Firecloud group
+   * @throws DuosDatasetNotFoundException if TDR has no record of the DUOS dataset
+   */
+  public DuosFirecloudGroupModel retrieveFirecloudGroup(String duosId) {
+    DuosFirecloudGroupModel firecloudGroup = duosDao.retrieveFirecloudGroupByDuosId(duosId);
+    if (firecloudGroup == null) {
+      throw new DuosDatasetNotFoundException(
+          "DUOS dataset %s has not been registered in TDR".formatted(duosId));
+    }
+    return firecloudGroup;
```

Comment:
Thoughts on renaming this method to `retrieveFirecloudGroupByDuosId`? I'm finding the existence of multiple versions of `retrieveFirecloudGroup` a little confusing.

Reply:
I appreciate the feedback, I can rename this.